x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.tsx

```diff
@@ -126,6 +130,36 @@ export const EndpointList = () => {
   const dispatch = useDispatch<(a: EndpointAction) => void>();
   // cap ability to page at 10k records. (max_result_window)
   const maxPageCount = totalItemCount > MAX_PAGINATED_ITEM ? MAX_PAGINATED_ITEM : totalItemCount;
+  const [showTransformFailedCallout, setShowTransformFailedCallout] = useState(false);
+
+  const getTransformStats = useCallback(async () => {
+    if (!services?.http) {
+      return;
+    }
+
+    const transformStats: TransformStatsResponse = await services.http.get(TRANSFORM_STATS_URL);
+    if (!transformStats?.transforms) {
```

Comment:
@pzl 

Thanks for pointing this out.  404 is a bit awkward with the current architecture, because you could run into the situation that the Endpoint package hasn't installed or upgraded yet.

I would want to be sure that we don't preemptively display a warning to the user while the Endpoint package is installing (i.e. no transform yet).

I'm OK with breaking out another ticket for the 404 case since this addresses the problem that we're seeing in practice where the transform is installed, but not running and the user doesn't know about it.

@pzl or @joeypoon can you write a ticket for the 404 case?

Reply:
created